### DISCUSSION TITLE
Scan of new deployment

### DISCUSSION THREAD

**Discussion by tommorgan365** [Security-Onion-Solutions/securityonion#1610] [Unsupported Versions] [answered]
Distributed environment 
manager
two search nodes
one fleet.
Deployed with your ISO.

Network Scan  is reporting  this on fleet:
Vulnerability					
ICMP timestamp response					
SSH Birthday attacks on 64-bit block ciphers (SWEET32)
SSH CBC vulnerability
SSH Server Supports 3DES Cipher Suite
SSH Server Supports diffie-hellman-group1-sha1
SSH Server Supports Weak Key Exchange Algorithms
TCP timestamp response

**Comment by TOoSmOotH** [comment, marked as answer]
ssh should be the latest and try and update every 15 minutes. Can you do a rpm -qa | grep ssh?

**Reply by tommorgan365** [reply]
# rpm -qa | grep ssh
openssh-clients-7.4p1-21.el7.x86_64
libssh2-1.8.0-3.el7.x86_64
openssh-7.4p1-21.el7.x86_64
openssh-server-7.4p1-21.el7.x86_64

**Reply by TOoSmOotH** [reply]
Did it say what CVE it was? That is the latest version of SSH

**Reply by tommorgan365** [reply]
No it did not Rapid7 scan,  The other machines showed a drop or reset in SO.
This is what it picked up on the fleet machine:
Port	Configuration	Version
22	OpenSSH	7.4
22	ssh.algorithms.compression	none,zlib@openssh.com
22	ssh.algorithms.encryption	chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com,aes128-cbc,aes192-cbc,aes256-cbc,blowfish-cbc,cast128-cbc,3des-cbc
22	ssh.algorithms.hostkey	ssh-rsa,rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519
22	ssh.algorithms.kex	curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1,diffie-hellman-group1-sha1
22	ssh.algorithms.mac	umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
22	ssh.banner	SSH-2.0-OpenSSH_7.4
22	ssh.protocol.version	2.0
22	ssh.rsa.pubkey.fingerprint	xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

**Reply by TOoSmOotH** [reply]
SSH should be open I don't understand the vulnerability here?

**Reply by B3DTech** [reply]
I believe what it's saying is that there are vulnerable SSH ciphers and weak key enabled that should be disabled.  And timestamp replies that should be blocked.

https://www.rapid7.com/db/vulnerabilities/ssh-3des-ciphers

https://www.rapid7.com/db/vulnerabilities/ssh-cbc-ciphers/

https://www.rapid7.com/db/vulnerabilities/ssh-cve-2015-4000/

https://www.rapid7.com/db/vulnerabilities/ssh-weak-kex-algorithms

https://www.rapid7.com/db/vulnerabilities/generic-tcp-timestamp/

https://www.rapid7.com/db/vulnerabilities/generic-icmp-timestamp/

**Reply by dougburks** [reply]
I've started a documentation page for ssh:
https://docs.securityonion.net/en/2.3/ssh.html

**Reply by tommorgan365** [reply]
Hello,

Do you need timestamp response on TCP and ICMP for SO ?  Cleared up the SSH issues.

**Reply by dougburks** [reply]
You should be able to disable those as well.

**Reply by tommorgan365** [reply]
Ok - TCP timestamp - host setting.  ICMP timestamp looks like it needs to be done at the firewall level at the fleet machine.  Any guidance on this one - read through firewall doc and want to be careful.

**Reply by B3DTech** [reply]
@dougburks  - So with your documentation you created - you're saying that by default, SecurityOnion, a security utility, will not by default make sure it's own services it exposes are secure and therefore leave itself open to vulnerabilities in weak ciphers and leaking info?

**Reply by dougburks** [reply]
I'm not saying that we won't ever make any changes here, just documenting current status.

**Reply by B3DTech** [reply]
Thanks for the follow up - the post/documentation came across very indignant.

**Reply by tommorgan365** [reply]
Meanwhile -- my question was ICMP timestamp looks like it needs to be done at the firewall level at the fleet machine. Any guidance on this one - read through firewall doc and want to be careful.

**Reply by dougburks** [reply]
We're not being indignant. We're simply trying to respond to as many questions and problems as quickly as possible given our limited resources. Thanks for your understanding.

**Reply by TOoSmOotH** [reply]
> Meanwhile -- my question was ICMP timestamp looks like it needs to be done at the firewall level at the fleet machine. Any guidance on this one - read through firewall doc and want to be careful.

I have put in a new issue #1704 for the ICMP thing.

**Reply by dougburks** [reply]
We've made some updates to https://docs.securityonion.net/en/2.3/ssh.html.  As always, please keep in mind that documentation is a work in progress.
